runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -79,13 +79,16 @@
 
   private static final String dockerContainerImageOption = "docker_container_image";
   private static final String externalServiceAddressOption = "external_service_address";
+  private static final String externalServiceExecutableOption = "external_service_executable";
```

Comment:
What is the executable expected to be? What benefit is there to allowing an arbitrary executable? We may want to simplify things for the user by making this a boolean (processes/threads) with a fixed executable otherwise.

Reply:
Good question, I had the same question for myself as well. 

I ended up putting configurable executable because I noticed this is the pattern in the Python external worker pool support [1]. However, I did notice that in boot script [2] itself, it hard codes executable to be `/opt/apache/beam/boot` when `--worker_pool` is specified. 

I suppose we may follow the same pattern as Python to make external worker pool itself configurable with executable but hard code `/opt/apache/beam/boot` in boot itself, alternatively we may just hard code in external worker pool itself. WDYT.

[1] https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/worker/worker_pool_main.py#L64
[2]  https://github.com/apache/beam/blob/master/sdks/python/container/boot.go#L91